service/src/main/java/bio/terra/workspace/service/workspace/WsmApplicationService.java

```diff
@@ -28,15 +34,79 @@
   private static final Logger logger = LoggerFactory.getLogger(WsmApplicationService.class);
 
   private final ApplicationDao applicationDao;
+  private final JobService jobService;
   private final WsmApplicationConfiguration wsmApplicationConfiguration;
+  private final WorkspaceService workspaceService;
+  // -- Testing Support --
+  // Unlike most code, the configuration code runs at startup time and does not have any output
+  // beyond writing to the log. In order to test it, we introduce a test mode and wrap the error
+  // logging. When test mode is enabled, the wrapper saves a string array of log messages in
+  // addition to calling the logger.
+  private boolean testMode = false;
+  private List<String> errorList;
 
   @Autowired
   public WsmApplicationService(
-      ApplicationDao applicationDao, WsmApplicationConfiguration wsmApplicationConfiguration) {
+      ApplicationDao applicationDao,
+      JobService jobService,
+      WsmApplicationConfiguration wsmApplicationConfiguration,
+      WorkspaceService workspaceService) {
     this.applicationDao = applicationDao;
+    this.jobService = jobService;
     this.wsmApplicationConfiguration = wsmApplicationConfiguration;
+    this.workspaceService = workspaceService;
   }
 
+  // -- REST API Methods -- //
+
+  public void disableWorkspaceApplication(
+      AuthenticatedUserRequest userRequest,
+      UUID workspaceId,
+      UUID applicationId,
+      String resultPath,
+      String jobId) {
+
+    workspaceService.validateWorkspaceAndAction(
+        userRequest, workspaceId, SamConstants.SAM_WORKSPACE_OWN_ACTION);
+
```

Comment:
Should this (and the other application methods) also have a call to `StageService.assertMcWorkspace` for now?

Reply:
Yes, I will add it.